### PR TITLE
Updated docs - filters, intro, support, layout

### PR DESCRIPTION
**Pull request by jonstrutz11**
Major Changes
- Added intro
- Added page to create filters
- Added metabolomics filter docs
- Updated support page

Minor Changes
- Created API reference top-level section
- Organized doc .rst files for API reference in a single folder
- Created issue template for repo (for support)

Note that we still have to add installation instructions to docs, but I wanted to wait until we packaged everything up.

### CODE REVIEW

File: docs/pickaxe_run.rst [hunk 1]
```diff
@@ -16,7 +16,8 @@ and explain the components of running pickaxe. Generally, pickaxe_run.py operate
 4. Specification of Filters
 
 This document gives the relevant code snippets from a template and expands on existing comments. Additionally, brief 
-examples of relevant inputs will be created. For more detailed descriptions please see :doc:`inputs` and :doc:`filters`.
+examples of relevant inputs will be created. For more detailed descriptions please see :doc:`inputs` and :doc:`api_ref/filters`.
+To create custom filters, see :doc:`custom_filters`.
```
Comment (KShebek) [inline docs/pickaxe_run.rst:20]: We could make this a tip to really draw attention

```rst
... tip::
    To create custom...
```

**Review by KShebek — CHANGES_REQUESTED** [review]

File: docs/index.rst [hunk 2]
```diff
@@ -14,10 +14,39 @@ MINE-Database Documentation
    command_line.rst
    pickaxe_run.rst
    inputs.rst
+   custom_filters.rst
    
-   pickaxe.rst
-   reactions.rst
-   metabolomics.rst
-   filters.rst
+   api_reference.rst
```
[the remark this answers is not in the document]
Comment (jonstrutz11) [reply docs/index.rst:19]: Yeah, the term is so vague that it doesn't really have clear boundaries anymore - I was just copying the language used for the cobrapy codebase.

[the remark this answers is not in the document]
Comment (jonstrutz11) [reply docs/index.rst:19]: https://cobrapy.readthedocs.io/en/latest/

File: docs/pickaxe_run.rst (hunk 1 above)
Comment (jonstrutz11) [reply docs/pickaxe_run.rst:20]: good idea - I'll add that in

**Review by KShebek — APPROVED** [review]
Looks good to me.

**Pull request merged**